scheduler/reconcile_util.go

```diff
@@ -411,6 +422,12 @@ func (a allocSet) filterByRescheduleable(isBatch, isDisconnecting bool, now time
 			continue
 		}
 
+		// Allocations with host volume IDs can be ignored, but we must keep
+		// the information they carry for future migrated allocs
+		if len(alloc.HostVolumeIDs) > 0 {
+			informational = append(informational, alloc)
+		}
```

Comment:
Do we need to make sure we're _only_ doing this for ignored / untainted allocations? For example, suppose we have `count=2` and 2 allocations: 1 is previously stopped and 1 just failed. How do we avoid having both the rescheduled allocation and the replacement `allocPlaceResult` from getting the same `previousAlloc` value?